### ISSUE TITLE
bug report: getting error messages during startup with SMTP_ONLY=1

### ISSUE THREAD

**Issue by jwaes** [docker-mailserver/docker-mailserver#3646]
### 📝 Preliminary Checks

- [X] I tried searching for an existing issue and followed the [debugging docs](https://docker-mailserver.github.io/docker-mailserver/latest/config/debugging/) advice, but still need assistance.

### 👀 What Happened?

Although i have set `SMTP_ONLY=1` in my env file, during setup i still get these error messages:

```
[   INF   ]  Configuring mail server
/usr/local/bin/setup.d/security/misc.sh: line 273: /usr/lib/dovecot/sieve-global/after/spam_to_junk.sieve: No such file or directory
error: script not found.
sievec: Fatal: failed to compile sieve script '/usr/lib/dovecot/sieve-global/after/spam_to_junk.sieve'
chown: cannot access '/usr/lib/dovecot/sieve-global/after/spam_to_junk.sieve': No such file or directory
chown: cannot access '/usr/lib/dovecot/sieve-global/after/spam_to_junk.svbin': No such file or directory
[   INF   ]  Starting daemons
```

As i understood, dovecot should not have started ?

### 👟 Reproduction Steps

just started it with a `sudo docker-compose up -d` 

### 🐋 DMS Version

[   INF   ]  Welcome to docker-mailserver 12.1.0

### 💻 Operating System and Architecture

Ubuntu 22.04.3 LTS x86_64

### ⚙️ Container configuration files

```yml
services:
  mailserver:
    image: ghcr.io/docker-mailserver/docker-mailserver:latest
    container_name: mailserver
    # Provide the FQDN of your mail server here (Your DNS MX record should point to this value)
    hostname: mail.xxxxxxx.net
    env_file: mailserver.env
    # More information about the mail-server ports:
    # https://docker-mailserver.github.io/docker-mailserver/latest/config/security/understanding-the-ports/
    # To avoid conflicts with yaml base-60 float, DO NOT remove the quotation marks.
    ports:
      - "25:25"    # SMTP  (explicit TLS => STARTTLS, Authentication is DISABLED => use port 465/587 instead)
      - "143:143"  # IMAP4 (explicit TLS => STARTTLS)
      - "465:465"  # ESMTP (implicit TLS)
      - "587:587"  # ESMTP (explicit TLS => STARTTLS)
      - "993:993"  # IMAP4 (implicit TLS)   
    volumes:
      - ./docker-data/dms/mail-data/:/var/mail/
      - ./docker-data/dms/mail-state/:/var/mail-state/
      - ./docker-data/dms/mail-logs/:/var/log/mail/
      - ./docker-data/dms/config/:/tmp/docker-mailserver/
      - /etc/localtime:/etc/localtime:ro
      - /etc/traefik/certs/acme-cloudflare.json:/etc/letsencrypt/acme.json:ro
    restart: always
    stop_grace_period: 1m
    # Uncomment if using `ENABLE_FAIL2BAN=1`:
    # cap_add:
    #   - NET_ADMIN
    healthcheck:
      test: "ss --listening --tcp | grep -P 'LISTEN.+:smtp' || exit 1"
      timeout: 3s
      retries: 0
```

### 📜 Relevant log output

```Text
[   INF   ]  Configuring mail server
/usr/local/bin/setup.d/security/misc.sh: line 273: /usr/lib/dovecot/sieve-global/after/spam_to_junk.sieve: No such file or directory
error: script not found.
sievec: Fatal: failed to compile sieve script '/usr/lib/dovecot/sieve-global/after/spam_to_junk.sieve'
chown: cannot access '/usr/lib/dovecot/sieve-global/after/spam_to_junk.sieve': No such file or directory
chown: cannot access '/usr/lib/dovecot/sieve-global/after/spam_to_junk.svbin': No such file or directory
[   INF   ]  Starting daemons
```

**Comment by georglauterbach** [comment]
I will have a look!

**Comment by georglauterbach** [comment]
Can you please check whether this also happens with the tag `:edge`? I had a look at the scripts and as far as I can tell, we guard the setup now (i.e. check whether `SMTP_ONLY=1`).

**Comment by jwaes** [comment]
@georglauterbach indeed, with the `:edge` tag it shows the expected behaviour.

**Comment by georglauterbach** [comment]
Good. We will release v13.0.0 soon, so you can update to that stable release. Please make sure to read the changeog thoroughly :) I will close this as you know what to do.

PS: Due to the way we structure our releases, we do not provide hotfixes like these for older releases. We're sorry for that, but the way we do releases is simple and not too heavy on maintainer's time, which is a real concern. That's why you won't see the fix on v12.1.0.
